### PR TITLE
Modified the tests to use mock functionality of get_child_port function under portconfig utility

### PR DESCRIPTION
**Pull request by samaity**
Signed-off-by: Sangita Maity <samaity@linkedin.com>

**What I did**
Modified the tests to use mock functionality of get_child_port function under portconfig utility.  This will remove the dependency of port config utility in the sonic-buildimage and make the test case more robust specific to the DPB cli utility.

**How I did it**
used MagicMock feature

**How to verify it**
All test cases are passed in config_dpb_test.py file
```

tests/config_dpb_test.py::TestConfigDPB::test_get_breakout_options PASSED [ 10%]
tests/config_dpb_test.py::TestConfigDPB::test_config_breakout_extra_table_warning PASSED [ 11%]
tests/config_dpb_test.py::TestConfigDPB::test_config_breakout_verbose PASSED [ 11%]
tests/config_dpb_test.py::TestConfigDPB::test_config_breakout_negative_cases PASSED [ 11%]
tests/config_dpb_test.py::TestConfigDPB::test_config_breakout_various_modes PASSED [ 11%]

```

### CODE REVIEW

File: tests/config_dpb_test.py [hunk 1]
```diff
@@ -16,7 +17,8 @@
     os.path.join(os.path.dirname(__file__), '..', 'config', 'config_mgmt.py'))
 import config_mgmt
 
-breakout_cfg_file_json = {
+# Sample platform.json for Test
+BRKOUT_CFG_FILE_JSON = {
```
Comment (zhenggen-xu) [inline tests/config_dpb_test.py:21]: remove below lines with  "alias_at_lanes".

Comment (samaity) [reply tests/config_dpb_test.py:21]: #Removed

**Comment by samaity** [comment]
retest this please

**Comment by samaity** [comment]
@jleveque -- Kindly review whenever possible. need help in merging the code. Thanks!

File: tests/config_dpb_test.py [hunk 2]
```diff
@@ -300,27 +417,46 @@ def test_config_breakout_negative_cases(self, sonic_db):
...
+
+
         # Wrong interface name
         result = runner.invoke(config.config.commands["interface"].\
-            commands["breakout"], ['Ethern', '2x50G', '-v', '-y'], obj=obj)
+            commands["breakout"], ['{}'.format(interface), '{}'.format(newMode), '-v', '-y'], obj=obj)
+
         print(result.exit_code, result.output)
         assert result.exit_code == 1
         #TODO: Uncomment it after Dev Fix, right now it is python bt
```
Comment (jleveque) [inline tests/config_dpb_test.py:432]: Looks like this is done with this commit. Remove this TODO comment?

File: tests/config_dpb_test.py [hunk 3]
```diff
@@ -362,99 +498,178 @@ def test_config_breakout_various_modes(self, sonic_db):
...
         def config_dpb_port8_2x50G_1x100G_f():
 
+            # Input Data
+            interface = 'Ethernet8'
+            curMode = '2x50G'
+            newMode = '1x100G[40G]'
+
+            print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+            config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:559]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 4]
```diff
@@ -362,99 +498,178 @@ def test_config_breakout_various_modes(self, sonic_db):
...
         def config_dpb_port8_1x100G_4x25G():
+
+            # Input Data
+            interface = 'Ethernet8'
+            curMode = '1x100G[40G]'
+            newMode = '4x25G[10G]'
+
+            print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+            config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:584]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 5]
```diff
@@ -362,99 +498,178 @@ def test_config_breakout_various_modes(self, sonic_db):
...
         def config_dpb_port8_4x25G_1x100G_f():
 
+            # Input Data
+            interface = 'Ethernet8'
+            curMode = '4x25G[10G]'
+            newMode = '1x100G[40G]'
+
+            print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+            config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:610]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 6]
```diff
@@ -362,99 +498,178 @@ def test_config_breakout_various_modes(self, sonic_db):
...
         def config_dpb_port8_1x100G_1x50G_2x25G_f_l():
 
+            # Input Data
+            interface = 'Ethernet8'
+            curMode = '1x100G[40G]'
+            newMode = '1x50G(2)+2x25G(2)'
+
+            print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+            config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:635]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 7]
```diff
@@ -362,99 +498,178 @@ def test_config_breakout_various_modes(self, sonic_db):
...
         def config_dpb_port4_4x25G_2x50G_f_l():
 
+            # Input Data
+            interface = 'Ethernet4'
+            curMode = '4x25G[10G]'
+            newMode = '2x50G'
+
+            print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+            config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:660]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 8]
```diff
@@ -229,27 +327,35 @@ def test_config_breakout_extra_table_warning(self, breakout_cfg_file, sonic_db):
...
+        newMode = '2x50G'
+
         # Mock functions except breakout_warnUser_extraTables
         mock_funcs[0] = config.device_info.get_path_to_port_config_file
         mock_funcs[1] = config.load_ConfigMgmt
         config.device_info.get_path_to_port_config_file = \
             mock.MagicMock(return_value = breakout_cfg_file)
         config.load_ConfigMgmt = \
             mock.MagicMock(return_value = config_mgmt_dpb(db.cfgdb))
+        config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:342]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 9]
```diff
@@ -275,8 +381,19 @@ def test_config_breakout_verbose(self, sonic_db):
...
         obj = {'config_db':db.cfgdb}
 
+        # Input Data
+        interface = 'Ethernet0'
+        curMode = '4x25G[10G]'
+        newMode = '2x50G'
+
+        print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+        config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:391]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 10]
```diff
@@ -362,99 +498,178 @@ def test_config_breakout_various_modes(self, sonic_db):
...
         def config_dpb_port8_4x25G_2x50G_f_l():
 
+            # Input Data
+            interface = 'Ethernet8'
+            curMode = '4x25G[10G]'
+            newMode = '2x50G'
+
+            print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+            config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:508]: Trailing backslash should not be necessary due to open parenthesis

File: tests/config_dpb_test.py [hunk 11]
```diff
@@ -362,99 +498,178 @@ def test_config_breakout_various_modes(self, sonic_db):
...
         def config_dpb_port8_2x50G_1x100G():
 
+            # Input Data
+            interface = 'Ethernet8'
+            curMode = '2x50G'
+            newMode = '1x100G[40G]'
+
+            print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
+
+            config.get_child_ports = mock.MagicMock( \
```
Comment (jleveque) [inline tests/config_dpb_test.py:533]: Trailing backslash should not be necessary due to open parenthesis

**Review by jleveque — CHANGES_REQUESTED** [review]

File: tests/config_dpb_test.py (hunk 11 above)
Comment (samaity) [reply tests/config_dpb_test.py]: #done

File: tests/config_dpb_test.py (hunk 2 above)
Comment (samaity) [reply tests/config_dpb_test.py]: Removed

**Review by jleveque — APPROVED** [review]

**Comment by samaity** [comment]
Thanks, @jleveque !!!

**Pull request merged**